CMakeLists.txt

```diff
@@ -0,0 +1,132 @@
+project(ATF)
+cmake_minimum_required(VERSION 3.11)
+
+# Enable PIC as static libraries may be linked to shared objects
+set (CMAKE_POSITION_INDEPENDENT_CODE TRUE)
+set (CMAKE_CXX_STANDARD 11)
+set (CMAKE_INCLUDE_CURRENT_DIR ON)
+set (CMAKE_AUTOMOC ON)
+
+find_package(Qt5 COMPONENTS Core Network WebSockets REQUIRED)
+find_package(Lua 5.2 EXACT REQUIRED)
+
+list(GET LUA_LIBRARIES 0 LUA_LIB)
+list(GET LUA_LIBRARIES 1 LIBM_LIB)
+
+message(STATUS "lualib : ${LUA_LIB}")
+message(STATUS "libmlib : ${LIBM_LIB}")
+
+add_library(lua::lua UNKNOWN IMPORTED GLOBAL)
+set_target_properties(lua::lua
+  PROPERTIES
+    INTERFACE_INCLUDE_DIRECTORIES "${LUA_INCLUDE_DIR}"
+    IMPORTED_LOCATION ${LUA_LIB}
+    IMPORTED_LINK_DEPENDENT_LIBRARIES ${LIBM_LIB}
+    )
+
+execute_process(COMMAND git submodule init WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR})
+execute_process(COMMAND git submodule update WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR})
+
+add_executable(${PROJECT_NAME}
+                src/network.cc
+                src/timers.cc
+                src/qtdynamic.cc
+                src/qtlua.cc
+                src/qdatetime.cc
+                src/marshal.cc
+                src/main.cc
+                src/lua_interpreter.cc
+                )
+
+target_link_libraries(${PROJECT_NAME}
+                       Qt5::Core
+                       Qt5::Network
+                       Qt5::WebSockets
+                       lua::lua
+                       )
+
+if(CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
+  message(INFO "CMAKE_CURRENT_BINARY_DIR: " ${CMAKE_CURRENT_BINARY_DIR})
+  set(CMAKE_INSTALL_PREFIX
+  ${CMAKE_CURRENT_BINARY_DIR} CACHE PATH "without install prefix" FORCE
+  )
+endif(CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
+
+set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_CURRENT_SOURCE_DIR}")
+set(CMAKE_SOURCE_PREFIX ${CMAKE_SOURCE_PREFIX} "${CMAKE_INSTALL_PREFIX}")
+find_package (BSON)
```

Comment:
@SNiukalov could you extract bson search to separate cmake file?

Reply:
@LuxoftAKutsan 
Fixed in: https://github.com/smartdevicelink/sdl_atf/pull/158/commits/e26ccfb337fed69c55707557f491fa8bc28d2b22